tests/test_qnode.py

```diff
@@ -111,6 +114,53 @@ def test_op_successors_extracts_all_successors(self, qnode):
         assert qnode.ops[4] in successors
         assert qnode.ops[5] not in successors
 
+    def test_print_applied(self, mock_device):
+        """Test that printing applied gates works correctly"""
+
+        H = np.array([[0, 1], [1, 0]])
+
+        def circuit(x):
+            qml.RX(x, wires=[0])
+            qml.CNOT(wires=[0, 1])
+            qml.RY(0.4, wires=[0])
+            qml.RZ(-0.2, wires=[1])
+            return qml.expval(qml.PauliX(0)), qml.var(qml.Hermitian(H, wires=1))
+
+        expected_qnode_print = textwrap.dedent("""\
+            Operations
+            ==========
+            RX({x}, wires=[0])
+            CNOT(wires=[0, 1])
+            RY(0.4, wires=[0])
+            RZ(-0.2, wires=[1])
+
+            Observables
+            ===========
+            expval(PauliX(wires=[0]))
+            var(Hermitian([[0 1]
+             [1 0]], wires=[1]))""")
+
+        node = qml.QNode(circuit, mock_device)
+
+        # test before construction
+        f = io.StringIO()
+
+        with contextlib.redirect_stdout(f):
+            node.print_applied()
+            out = f.getvalue().strip()
+
+        assert out == "QNode has not yet been executed."
```

Comment:
One could think of having this in a separate test, but I think two assert statements in one test are fine.

Reply:
Trying to keep it contained for now, as it will need to be moved to the new QNode :laughing: